memory_engine/memory_store.py

```
@@ -233,81 +171,63 @@ def replay(self, start_time: datetime, end_time: datetime) -> List[Dict]:
-                return False
+        """Load memory from disk with validation and error handling."""
+        try:
+            # Security: Validate storage path is within base directory (prevents path traversal)
+            resolved_path = os.path.abspath(self.storage_path)
+            if not resolved_path.startswith(MEMORY_STORE_BASE_DIR):
+                logger.error(
+                    f"⚠️  Storage path traversal attempt blocked: {self.storage_path}"
+                )
+                raise ValueError(
+                    f"Storage path must be within {MEMORY_STORE_BASE_DIR}"
+                )
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)